template/src/services/datareader.js

```diff
@@ -83,10 +85,11 @@ export async function readPaginatedData (url) {
 /**
  * Requests data from a given url, resolving to response.data (default) or any other optionally specified value
  * @param url
+ * @param headers
  * @param resolve
  * @returns {Promise<*>}
  */
-export async function readData (url, resolve = d => d.data) {
-  let response = await get(url)
+export async function readData (url, headers = null, resolve = d => d.data) {
+  let response = await (headers ? get(url, headers) : get(url))
   return resolve(response)
```

Comment:
Same here, the null value is handled by get and the logic should not be repeated